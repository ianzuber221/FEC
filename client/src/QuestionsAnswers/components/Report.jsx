import React, { useState } from 'react';
import styled from 'styled-components';
import { updateReport } from '../lib/api/githubAPI';

function Report({ id, type }) {
  const [reported, setReported] = useState(false);
  const handleClick = () => {
    if (reported) {
      return;
    }
    const executed = confirm("Are you sure you want to report this?");
    if (executed) {
      updateReport(type, id)
        .then(() => {
          setReported(true);
        });
    }
  };

  return (
    <ClickableU onClick={handleClick} onKeyDown={handleClick} role="button" tabIndex="-1">
      {reported ? 'Reported' : 'Report'}
    </ClickableU>
  );
}

export default Report;

const ClickableU = styled.u`
  &:hover {
    cursor: pointer;
    opacity: 60%;
  }
`;