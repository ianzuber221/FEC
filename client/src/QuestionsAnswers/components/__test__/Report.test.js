/**
 * @jest-environment jsdom
 */
/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { jest } from '@jest/globals';
import Report from '../Report';
import '@testing-library/jest-dom/extend-expect';

let confirmSpy;


beforeAll(() => {
    confirmSpy = jest.spyOn(window, 'confirm');
    confirmSpy.mockImplementation(jest.fn(() => false));
});

afterAll(() => confirmSpy.mockRestore());

it('should match snapshot', async () => {
  const component = renderer.create(
    <Report
      id={5986818}
      type='answer'
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should appear a confirmation message when report is clicked', async () => {
  const { findByText, getByText } = render(
    <Report
      id={5986821}
      type='answer'
    />
  );

  const report = getByText('Report');
  await userEvent.click(report);
  expect(confirmSpy).toBeCalled();
});