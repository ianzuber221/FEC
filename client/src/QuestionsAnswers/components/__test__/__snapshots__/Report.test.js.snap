// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<u
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="-1"
>
  Report
</u>
`;
